### ISSUE TITLE
Editing lua code results in error due to missing linter

### ISSUE THREAD

**Issue by mcctuxic** [NTBBloodbath/doom-nvim#125]
When editing the configuration section for options in doom_config.lua I did a typo.
doomnvim on-the-fly checker tries (?) to detect it when writing the file and failed with the following message:
![noLuaLint](https://user-images.githubusercontent.com/30279786/134279641-d46eac3a-0e2c-4498-9902-270817fea763.jpg)

The linter is activated in doom_modules.lua.

**Comment by NTBBloodbath** [comment]
Hi, sorry for the late reply.

Our current linting setup automatically adds linters for current language if supported by the linter plugin, I think we should add a new option to set "Hey, I want this language to have X linter" instead of having all the available ones enabled because maybe the end user doesn't have them installed. Should we do this in our `doom_config.lua` or as an extra field in the languages on `doom_modules.lua`?

The difference between both would be the following

> doom_config.lua

```lua
linters = {
	lua = { "luacheck" },
	cpp = { "clangtidy" },
},
```

> doom_modules.lua

```lua
langs = {
	"lua +lsp +linter",
	"cpp +linter +debug",
},
```

Please let me know what one do you think could be better,

Regards

**Comment by mcctuxic** [comment]
Hi,

no need to be sorry - I am happy to have doomnvim installed! :)

Generally speaking I think putting something into a configuration, which
cannot detected as missing (other than with low level error messages) 
and cannot be added by doomnvim via Packer for example automagically
should not be there.

It is better to have a configuration which works and does not have a 
certain feature (linter in this case) as a configuration, which
supports a feature, which is not installed and needs to be installed by
the user.

With the suggested addition to the configuration exactly this is acchieved!

Plus Bonus: In case a certain linter of a certain version does not work,
now it will be possible to switch off that linter granually.

I think it will help new "Jedis of the NVIM" to recognize the
possibilities, if that configuration (that is language + lsp + linter)
will be there in the configuration by default but being commented
out/off (sorry, no native speaker).

Long story short: Yes! Great idea, NTBBloodbath!

**Comment by NTBBloodbath** [comment]
Hey, this should be fixed now. I chose the `doom_config.lua` approach, I found it easier for our end users, please let me know if you find a bug :)
